File: lab-3/Company.cs
using System.Collections.ObjectModel;

public abstract class Company
{
    public int CompanyID { get; set; }
    public string? CompanyName { get; set; }
    public string? Street { get; set; }    
    public string? City { get; set; }
    public string? ZipCode { get; set; } 

    // Pole do identyfikacji typu
    public string CompanyType { get; set; }   
}

File: hibernate-lab/src/main/java/org/example/model/Company.java
package org.example.model;

import jakarta.persistence.*;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public class Company {
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    private Long companyId;
    private String companyName;
    private String street;
    private String city;
    private String zipCode;

    public Company() {}

    public Company(String companyName, String street, String city, String zipCode) {
        this.companyName = companyName;
        this.street = street;
        this.city = city;
        this.zipCode = zipCode;
    }
}
